src/main/java/neqsim/thermodynamicoperations/flashops/SolidFlash12.java

```diff
@@ -143,19 +143,19 @@ public double calcQ() {
     for (int i = 0; i < solidsNumber; i++) {
       Q += system.getPhase(0).getComponent(solidIndex).getz()
           * (1 - Math.log(system.getPhase(0).getComponent(solidIndex).getz()
-              / system.getPhases()[3].getComponent(solidIndex).getFugacityCoefficient()));
+              / system.getPhase(3).getComponent(solidIndex).getFugacityCoefficient()));
       for (int j = 0; j < system.getNumberOfPhases() - solidsNumber; j++) {
         Q -= system.getBeta(j)
-            * system.getPhases()[3].getComponent(solidIndex).getFugacityCoefficient()
+            * system.getPhase(3).getComponent(solidIndex).getFugacityCoefficient()
             / system.getPhase(j).getComponent(solidIndex).getFugacityCoefficient();
       }
     }
 
     for (int k = 0; k < system.getNumberOfPhases() - solidsNumber; k++) {
       dQdbeta[k] = 1.0;
-      for (int i = 0; i < system.getPhases()[0].getNumberOfComponents(); i++) {
+      for (int i = 0; i < system.getPhase(0).getNumberOfComponents(); i++) {
         if (i == solidIndex) {
-          dQdbeta[k] -= system.getPhases()[3].getComponents()[solidIndex].getFugacityCoefficient()
+          dQdbeta[k] -= system.getPhase(3).getComponents()[solidIndex].getFugacityCoefficient()
```

Comment:
In this case I beleive we have to used system.getPhases()[3] - as the solid phase is always number three (and getPhase(3) can point to a different phase.